packages/amplify-graphql-api-construct-tests/src/utils/sql-provider-helper.ts

```diff
@@ -0,0 +1,131 @@
+          return;
+        }
+        const tableName = getMappedName(node);
+        const fieldStatements = [];
+        const fieldsToAdd = node.fields.filter((field) => !isRelationalField(field));
+        fieldsToAdd.forEach((field, index) => {
+          fieldStatements.push(getFieldStatement(field, index === 0, engine));
+        });
+        const sql = `CREATE TABLE ${convertToDBSpecificName(tableName, engine)} (${fieldStatements.join(', ')});`;
+        sqlStatements.push(sql);
+      },
+    },
+  };
+  visit(document, schemaVisitor);
+  return sqlStatements;
+};
+
+const getMappedName = (definition: ObjectTypeDefinitionNode | FieldDefinitionNode): string => {
+  const name = definition?.name?.value;
+  const refersToDirective = definition?.directives?.find((d) => d?.name?.value === 'refersTo');
+  if (!refersToDirective) {
+    return name;
+  }
+  const mappedName = (refersToDirective?.arguments?.find((a) => a?.name?.value === 'name')?.value as StringValueNode)?.value;
+  if (!mappedName) {
+    return name;
+  }
+  return mappedName;
+};
+
+const isRelationalField = (field: FieldDefinitionNode): boolean => {
+  return field?.directives?.some((d) => [HAS_MANY_DIRECTIVE, HAS_ONE_DIRECTIVE, BELONGS_TO_DIRECTIVE].includes(d?.name?.value));
+};
+
+const getFieldStatement = (field: FieldDefinitionNode, isPrimaryKey: boolean, engine: ImportedRDSType): string => {
+  const fieldName = getMappedName(field);
+  const fieldType = field.type;
+  const isNonNull = fieldType.kind === Kind.NON_NULL_TYPE;
+  const baseType = getBaseType(fieldType);
+  const columnType = isArrayOrObject(fieldType, []) ? getArrayOrObjectFieldType(fieldType, engine) : convertToSQLType(baseType);
+  // Check if @default is defined on field
+  const defaultDir = field.directives.find((dir) => dir.name.value === 'default');
+  const defaultValueNode = defaultDir?.arguments.find((arg) => arg.name.value === 'value');
+  const fieldDefaultValue = defaultDir && defaultValueNode ? valueFromASTUntyped(defaultValueNode.value) : undefined;
+  const sql = `${convertToDBSpecificName(fieldName, engine)} ${columnType} ${isNonNull ? 'NOT NULL' : ''} ${
+    isPrimaryKey ? 'PRIMARY KEY' : ''
+  } ${fieldDefaultValue ? `DEFAULT ${fieldDefaultValue}` : ''}`;
+  return sql;
+};
+
+const getArrayOrObjectFieldType = (fieldType: TypeNode, engine: ImportedRDSType): string => {
+  switch (engine) {
+    case ImportedRDSType.MYSQL:
+      return 'JSON'; // MySQL does not support array types
+    case ImportedRDSType.POSTGRESQL:
+      return isListType(fieldType) ? 'VARCHAR[]' : 'json';
+    default:
+      return 'VARCHAR[]';
+  }
+};
+
+const convertToSQLType = (type: string): string => {
+  switch (type) {
+    case 'ID':
+    case 'String':
+      return 'VARCHAR(255)';
+    case 'Int':
+      return 'INT';
+    case 'Float':
+      return 'FLOAT';
+    case 'Boolean':
+      return 'BOOLEAN';
+    case 'AWSDateTime':
+      return 'DATETIME';
+    default:
+      return 'VARCHAR(255)';
```

Comment:
nit: un-recognized type can just throw an error here, so we know how to handle those.

Reply:
now throw error in this case